[Bug] Error with arrays and ValidatorFunctions

Hi,

I've got a 'People' model which is simply an array of persons:
```
People:
  type: array
  items:
    $ref: "#/components/schemas/Person"
```

This generates `export const PeopleValidators: {[K in keyof People]: Array<ValidatorFn>} = { };`, which results in the following transpilation error:

`ERROR in src/app/core/generated/api/model/people.ts(30,14): error TS2322: Type '{}' is not assignable to type '{ [x: number]: ValidatorFn[]; length: ValidatorFn[]; toString: ValidatorFn[]; toLocaleString: ValidatorFn[]; pop: ValidatorFn[]; push: ValidatorFn[]; concat: ValidatorFn[]; join: ValidatorFn[]; reverse: ValidatorFn[]; ... 21 more ...; includes: ValidatorFn[]; }'.
  Property 'length' is missing in type '{}'.`

I don't know if simply assigning `null` instead of empty object conflicts with the form-control-factory, so I didn't try..

I assume that this error will occur with every object that doesn't have own properties.

--- a/src/base-form-control-factory.ts
+++ b/src/base-form-control-factory.ts
@@ -21,7 +21,7 @@ export class BaseFormControlFactory<T> {
    *
    * @param validators An array of validators.
    */
-  constructor(model: T, validators: { [K in keyof T]: Array<ValidatorFn> }) {
+  constructor(model: T, validators: { [K in keyof T]?: Array<ValidatorFn> }) {
     this.map = new Map();
 
     for (const property in model) {
@@ -31,7 +31,7 @@ export class BaseFormControlFactory<T> {
 
       this.map.set(property, {
         value: model[property],
-        validators: validators[property],
+        validators: validators[property] ? validators[property] : [],
       });
     }
   }

--- a/src/mustache/modelGeneric.mustache
+++ b/src/mustache/modelGeneric.mustache
@@ -16,7 +16,7 @@ export interface {{classname}} {{#parent}}extends {{{parent}}} {{/parent}}{
 /**
  * A map of `ValidatorFn` for each property of {{classname}}.
  */
-export const {{classname}}Validators: {[K in keyof {{classname}}]: Array<ValidatorFn>} = {
+export const {{classname}}Validators: {[K in keyof {{classname}}]?: Array<ValidatorFn>} = {
 {{#vars}}
     {{name}}: [
         {{#required}}

